enclave_build/src/storage.rs

```diff
@@ -59,4 +71,312 @@ impl OciStorage {
             config: None,
         })
     }
+
+    /// Stores the image data provided as argument in the storage at the folder pointed
+    /// by the 'root_path' field.
+    pub fn store_image_data(&mut self, image_name: &str, image_data: &ImageData) -> Result<()> {
+        // Create the folder where the image data will be stored. Each image blob will be stored in
+        // a file named by the SHA256 digest of the content.
+        let blobs_path = self.root_path.join(STORAGE_BLOBS_FOLDER);
+        fs::create_dir_all(&blobs_path).map_err(EnclaveBuildError::OciStorageStoreError)?;
+
+        // Each layer file will be named after the layer's digest hash
+        Self::store_layers(&blobs_path, image_data)?;
+
+        // Store the manifest
+        let manifest = Self::store_manifest(&blobs_path, image_data)?;
+
+        // Store the config and validate UTF8 bytes
+        let config_content = Self::store_config(&blobs_path, image_data)?;
+
+        // If index file present, read and append new image entry to the JSON list
+        self.store_index(image_name, manifest)?;
+
+        // Write oci_layout file from template constant
+        self.store_layout()?;
+
+        self.config = Some(deserialize_from_reader(config_content.as_bytes())?);
+
+        Ok(())
+    }
+
+    fn default_oci_index() -> OciImageIndex {
```

Comment:
Can't we implement Default instead?

Reply:
This is the default index we use in this storage, not the default of the structure itself.